### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by snigam3112**

### CODE REVIEW

File: src/BayesianModel.py [hunk 1]
```diff
@@ -52,77 +63,141 @@ def add_edges(self, tail, head):
...
-        _order = list()
-        for user_given_state in states:
-            for state in self.node[node]['_states']:
-                if state[0] == user_given_state:
-                    _order.append(self.node[node]['_states'].index(state))
-                    break
-        self.node[node]['_rule_for_states'] = tuple(_order)
+        if self._all_states_mentioned(node, states):
+            _order = list()
+            for user_given_state in states:
```
Comment (ankurankan) [inline src/BayesianModel.py:138]: We can avoid a level of nesting here by using itertools.product.

File: src/BayesianModel.py [hunk 2]
```diff
@@ -147,40 +222,39 @@ def set_cpd(self, node, cpd):
...
-                             of states that are observed
-        @return:
+        observations: dictionary with key as node and value as the
+                      states that is observed
+        reset: if reset is True, the observation status is reset
         """
         #TODO check if multiple states of same node can be observed
-        #TODO if not above then, put validation
+        #TODO if above then, change code accordingly
         for _node in observations:
```
Comment (snigam3112) [inline src/BayesianModel.py:240]: Will look into it

File: src/BayesianModel.py [hunk 3]
```diff
@@ -52,77 +63,141 @@ def add_edges(self, tail, head):
...
-        _ord = list()
-        for user_given_parent in parents:
-            for parent in self.node[node]['_parents']:
-                if parent == user_given_parent:
-                    _ord.append(self.node[node]['_parents'].index(parent))
-                    break
-        self.node[node]['_rule_for_parents'] = tuple(_ord)
+        if self._all_parents_mentioned(node, parents):
+            _order = list()
+            for user_given_parent in parents:
```
Comment (ankurankan) [inline src/BayesianModel.py:188]: itertools.product

File: src/BayesianModel.py (hunk 2 above)
Comment (ankurankan) [inline src/BayesianModel.py:240]: Here we can't replace the loops by `iteltools.product`. I have replaced the loops in the other two lines.

**Pull request merged**